frontend/server/src/main/java/org/pytorch/serve/wlm/ModelManager.java

```diff
@@ -211,15 +213,38 @@ private void setupModelDependencies(Model model)
                             + requirementsFilePath; // NOPMD
 
             String[] envp =
-                    EnvironmentUtils.getEnvString(configManager.getModelServerHome(), null, null);
+                    EnvironmentUtils.getEnvString(
+                            model.getModelDir().getAbsolutePath(), null, null);
+
             Process process =
                     Runtime.getRuntime()
                             .exec(
                                     packageInstallCommand,
                                     envp,
                                     model.getModelDir().getAbsoluteFile());
+
             int exitCode = process.waitFor();
+
             if (exitCode != 0) {
+
+                String line;
+                String outputString = "";
+                // process's stdout is InputStream for caller process
+                BufferedReader brdr =
+                        new BufferedReader(new InputStreamReader(process.getInputStream()));
+                while ((line = brdr.readLine()) != null) {
+                    outputString += line;
+                }
+                String errorString = "";
+                // process's stderr is ErrorStream for caller process
+                brdr = new BufferedReader(new InputStreamReader(process.getErrorStream()));
+                while ((line = brdr.readLine()) != null) {
+                    errorString += line;
+                }
+
+                logger.info("Dependency installation stdout:\n" + outputString);
+                logger.error("Dependency installation stderr:\n" + errorString);
+
                 throw new ModelException(
                         "Custom pip package installation failed for " + model.getModelName());
```

Comment:
NIT: Consider appending the error trace to this exception instead

Reply:
This causes unit tests such as https://github.com/pytorch/serve/blob/master/frontend/server/src/test/java/org/pytorch/serve/ModelServerTest.java#L999 to fail, since they've been written for a specific package. The pip installation failure log in case of a package is a long trace that could be different most of the times. I'd like to retain the test because it's useful in it's current form, while adding the failure as an error log.

Reply:
We can append "Custom pip package installation failed for" to the error message and change the unit test to assert string contains instead of string equals